app/controllers/users/mfa_selection_controller.rb

```diff
@@ -13,16 +13,29 @@ def index
       analytics.user_registration_2fa_additional_setup_visit
     end
 
+    def non_restricted
+      @two_factor_options_form = TwoFactorOptionsForm.new(current_user)
+      @presenter = two_factor_options_presenter
+      render 'index'
+    end
```

Comment:
this seems to do everything inside the `index` method... except the analytics event log line. What if we just linked to index instead with a query param, like `?non_restricted=true`?

Reply:
What if we made that conditional?

```ruby
before_action :confirm_user_authenticated_for_2fa_setup, if: proc { !params[:non_restricted].to_s != 'true' }
```
or something